packages/java/endpoint/src/main/java/com/vaadin/hilla/signals/core/EventQueue.java

```diff
@@ -0,0 +1,124 @@
+package com.vaadin.hilla.signals.core;
+
+import jakarta.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+import reactor.core.publisher.Sinks.Many;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.locks.ReentrantLock;
+
+public abstract class EventQueue<T extends StateEvent> {
```

Comment:
Why is an event queue needed for tracking a single number value? A queue will eventually be needed for efficient resynch of more complex data structures but the most efficient way of catching up for a single value is always to just load the latest value.

Reply:
Looking at all types of Signals (no matter a single value holding or more complex ones such as list signals) simplified the early implementation of how we register signal instances to the SignalRegistry and how we receive updates (from client-side code) and dispatch them to the server-side instance of the shared signal. 
But, your point makes complete sense when it comes to the unnecessary overhead of backing single value holding signals with event queues, even if we initialize the size of the backing queue to one.

Reply:
It is easy enough to refactor and take out the event queue and the signal queue from this PR.